Fix a bug where ScrollableGrid/ScrollableList would flicker when

mpcomplete opened the pull request:
navigating away.

Details are in the bug, but when we navigate away, the overscroll
indicator around the Scrollable would change the widget hierarchy. This
would trigger the Scrollable to rebuild, which would cause its
scrollOffset to be clamped to 0 for a frame.

BUG=https://github.com/flutter/flutter/issues/4597

HansMuller commented:
LGTM

Hixie commented on packages/flutter/lib/src/material/overscroll_indicator.dart:
```diff
@@ -226,10 +226,11 @@ class _OverscrollIndicatorState extends State<OverscrollIndicator> {
-            return child;
+          // We keep the same widget hierarchy here, even when we're not
+          // painting anything, to avoid rebuilding the children. This
+          // fixes https://github.com/flutter/flutter/issues/4597 .
```
i wouldn't bother with the link to the closed bug. if people want the link they can always find it by looking at the history.
If there's something to say here we should say it inline, rather than point to the issue tracker.

Hixie commented:
LGTM, but we really should do the second fix in your bug on the long term. Can you add a `TODO(ianh)` or file a bug so that we remember to look at this at some point? I suspect this might become moot with #4430, so it's not something I'd worry about on the short term.

mpcomplete committed:
Fix a bug where ScrollableGrid/ScrollableList would flicker when
navigating away.

Details are in the bug, but when we navigate away, the overscroll
indicator around the Scrollable would change the widget hierarchy. This
would trigger the Scrollable to rebuild, which would cause its
scrollOffset to be clamped to 0 for a frame.

BUG=https://github.com/flutter/flutter/issues/4597
